### PR TITLE
save temporary tutorial files in the data folder

### LINKED ISSUE #757: make tutorial write files to a temporary folder that is listed in .gitignore
Currently, the tutorial writes the temporary files in the top-level pyuvdata folder when `doctest` is run. They are routinely deleted at the end of the doctest, but if doctest doesn't finish properly the files are not deleted and can be inadvertently added to the repo.

Instead, the files should be written to a temporary directory that is listed in .gitignore to prevent accidental addition of the tutorial files to the repo.

### PR DESCRIPTION
**Pull request by bhazelton**
## Description

Save temporary tutorial files in the `pyuvdata/data/tutorial_output` folder so they are covered by the .gitignore and don't contaminate the top level pyuvdata directory.
Also fix a couple of formatting issues and Sphinx warnings.

## Motivation and Context

closes #757 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation change (documentation changes only)
- [ ] Version change
- [ ] Build or continuous integration change

## Checklist:

- [x] I have read the [contribution guide](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/master/.github/CONTRIBUTING.md).
- [x] My code follows the code style of this project.

Documentation change checklist:
- [ ] Any updated docstrings use the [numpy docstring format](https://numpydoc.readthedocs.io/en/latest/format.html).
- [x] If this is a significant change to the readme or other docs, I have checked that they are rendered properly on ReadTheDocs. (you may need help to get this branch to build on RTD, just ask!)

### CODE REVIEW

**Comment by mkolopanis** [comment]
Want to drop this here so we don't forget. Pytest can wrap doctest http://doc.pytest.org/en/latest/doctest.html
we can use this to our advantage and make a conftest for the docs folder and make then delete a special folder regardless of whether the tutorial finishes.

**Comment by mkolopanis** [comment]
something interesting I noticed while running the tutorial myself. The order of the outputs of list(set(zip())) in line 507 changes in python 3.8. Not really sure why it does this just wanted to point it out.

```
(py3) matthew@bahamut:~$ ipython
Python 3.7.5 (default, Oct 25 2019, 15:51:11) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.10.2 -- An enhanced Interactive Python. Type '?' for help.

In [1]: list(set(zip([0,0,0],[21,2,6])))                                                                 
Out[1]: [(0, 6), (0, 21), (0, 2)]

In [2]: exit                                                                                                                                                                                                        
(py3) matthew@bahamut:~$ conda activate py38
(py38) matthew@bahamut:~$ ipython
Python 3.8.1 (default, Jan  8 2020, 22:29:32) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.11.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: list(set(zip([0,0,0],[21,2,6])))                                                                                                                                                                            
Out[1]: [(0, 2), (0, 21), (0, 6)]
```

File: docs/tutorial.rst [hunk 1]
```diff
@@ -1723,11 +1851,5 @@ Tutorial Cleanup
 -----------------
```
Comment (adampbeardsley) [inline docs/tutorial.rst]: Is this section needed now?

Comment (bhazelton) [reply docs/tutorial.rst]: probably not, I'll take it out.

**Comment by bhazelton** [comment]
I'm not sure about the changelog, I guess my bias is that it doesn't need to be updated for this but I can add something if you think I should.

**Pull request merged**